src/data-workspace/has-comment/has-comment-context.js

```diff
@@ -0,0 +1,32 @@
+import { useContext, createContext } from 'react'
+
+export const HasCommentContext = createContext({
```

Comment:
I'm a bit confused by the need for a context for this. Is it to prevent "subscribing" to `dataValues`?
I haven't looked too much into this, and I reckon you have a good reason for doing it like this. But to me it seems like this is something we should be able to compute pretty easily.

And to me it seems any change to the comment will re-render all fields regardless, since they subscribe to this context?

Reply:
And if it really is because we cannot reach into the dataValues in a performant way, I think its time to introduce something like https://github.com/pmndrs/zustand . 
I don't like that we have to keep the `dataValues` and this context in sync.

Reply:
Yeah I wasn't too happy with the implementation I ended up with, but there were some reasons why I resorted to it. I've left this PR a bit too long so I'm not a 100% sure if memory is serving me correctly, but I've checked a few things again and believe the following to be the case:
- The comment field is included the `dataValueSet` items in the `DataWorkSpace`, but that object changes a lot, i.e. whenever a field is changed and then blurred.
- The `DataEntryField` does not have direct access to its comment, and currently the data-entry fields don't all re-render every time the `dataValueSet` object changes. This is obviously behaviour we don't want to change.
- As such, deriving the `hasComment` value from the `dataValueSet` is problematic. The only way I could think of to avoid deriving the comment status from the `dataValueSet` was to take a more imperative approach which gives us much more granular control of when the `hasComment` values are updated.

So yeah, the problem was basically as you have already suggested in your second comment: we can't use `dataValueSet`. I also agree with your dislike to the fact that we will essentially be storing duplicate/overlapping information in two places and will have to ensure this will match. But to me this is the core issue and personally I can't think of away around this, while keeping re-renders limited. From what I've seen, if we were to use the `final-form-set-field-data` package you linked to, we'd still have to imperatively update the field's metadata right? If this is the case, I think adding this additional package would not really add much benefit over using a context....

Perhaps I am failing to see something here.....